File: Two-Pointers/moveZeros.py
def moveZeros(arr):
    i=0
    for j in range(len(arr)):
        if arr[j]==0:
            arr[i] = arr[j]
            i+=1
    return arr





arr = [1,0,2,0,3,0,4,0,5]
moveZeros(arr)
print(arr)

File: moveZeros.cpp
#include<bits/stdc++.h>
#include<vector>
using namespace std;

//naive approach -> O(N)

void pushZeros(vector<int>arr){
    int n = arr.size();
    vector<int>nonZeros(n);
    int j = 0;
    for(int i =0;i<n;i++){
        if(arr[i]!=0){
            nonZeros[j++] = arr[i];
        }
    }

}

int main(){
    int arr[]={1,2,0,4,3,0,5,0};
    int n = 8;
    moveZeros(arr,n);
}